modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java

```diff
@@ -839,7 +836,12 @@ else if (cmdH2 instanceof GridSqlDropTable) {
                 else {
                     ctx.security().authorize(tbl.cacheName(), SecurityPermission.CACHE_DESTROY);
 
+                    String schema = tbl.getSchema().getName();
+                    String tblName = tbl.getName();
+
                     ctx.query().dynamicTableDrop(tbl.cacheName(), cmd.tableName(), cmd.ifExists());
+
+                    schemaMgr.afterDropTable(schema, tblName);
```

Comment:
Looks like a wrong place to do all the internal stuff. Can we place it somewhere in SchemaManager::dropTable()?